In this project, I am a data analyst that works for a company that builds android and iOS mobile apps. These apps are made available on Google Play and in thr App Store. The company generates revenue through in-app adds since the apps are free to download and install. The more the users of a app, the better the revenue.


The goal of this project is to analyze data to help the company's devlopers understand what type of apps are likely to attract more users.

In [1]:
from csv import reader
open_google_play = open('googleplaystore.csv')
read_google_play = reader(open_google_play)
Android = list(read_google_play)

open_app_store = open('AppleStore.csv')
read_app_store = reader(open_app_store)
iOS = list(read_app_store)

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [3]:
google_play_data = explore_data(Android[1:], 0, 5, True)

app_store_data = explore_data(iOS[1:], 0, 5, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


Number of rows: 10841
Number of columns: 13
['284882215', 'Facebook', '38987980

In [4]:
print(Android[0])
print('\n')
print(iOS[0])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


In [5]:
print(Android[10473])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [6]:
del(Android[10473])

Looking through the android dataset, duplicate entries can be observed. Below is an example:

In [7]:
duplicate_entries = []
unique_entries = []
for row in Android[1:]:
    name = row[0]
    if name in unique_entries:
        duplicate_entries.append(name)
    else:
        unique_entries.append(name)

print('Number of unique entries:', len(unique_entries))
print('\n')
print('Number of duplicate entries:', len(duplicate_entries))
print('\n')
print('Examples of duplicate apps:', duplicate_entries[:10])

Number of unique entries: 9659


Number of duplicate entries: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack']


In the above code, it can be observed that only  9660 apps were unique while 1181 apps were recorded multiple times.

In [8]:
for app in Android[1:]:
    name = app[0]
    if name == 'ZOOM Cloud Meetings':
        print(app)
        
# This code is an example of a multiple entry. The 'ZOOM Cloud Meetings' app is entered two times.

['ZOOM Cloud Meetings', 'BUSINESS', '4.4', '31614', '37M', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 20, 2018', '4.1.28165.0716', '4.0 and up']
['ZOOM Cloud Meetings', 'BUSINESS', '4.4', '31614', '37M', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 20, 2018', '4.1.28165.0716', '4.0 and up']


There's a need to get rid of these duplicate entries. To do this, for any app with multiple entries, I will be keeping the entry with highest number of reviews while discarding the others.This is done since having the highest number of reviews signifies that the entry is the latest entry.

In the example above, the above critirion cannot be used since the number of reviews for the two entries are the same. Zoom Cloud Meetings have two entries with the same values for all columns, i will therefore be keeping one and discarding the other.

In [9]:
reviews_max = {}
for apps in Android[1:]:
    name = apps[0]
    n_reviews = float(apps[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    if name not in reviews_max:
        reviews_max[name] = n_reviews

print(len(reviews_max))

9659


In the above code, I created a dictionary showing all the unique apps with the highest number of reviews for each of them. The app names serve as the keys.

In [10]:
android_clean = []
already_added = []
for apps in Android[1:]:
    name = apps[0]
    n_reviews = float(apps[3])
    if n_reviews == reviews_max[name] and name not in already_added:
        android_clean.append(apps)
        already_added.append(name)
print(len(android_clean))
print(android_clean[:5])

9659
[['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up'], ['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up'], ['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up'], ['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up'], ['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']]


In the code above, I extracted a clean dataset from the original dataset. No duplicates exist.

In [11]:
def english_checker(a_string):
    non_ascii = 0
    for char in a_string:
        if ord(char) > 127:
            non_ascii += 1
    if non_ascii > 3:
        return False
    else:
        return True
print(english_checker('Instagram'))
print(english_checker('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(english_checker('Docs To Go™ Free Office Suite'))
print(english_checker('Instachat 😜'))

True
False
True
True


In [12]:
Android_english = []
iOS_english = []

for app in android_clean:
    name = app[0]
    if english_checker(name):
        Android_english.append(app)
        
for app in iOS[1:]:
    name = app[1]
    if english_checker(name):
        iOS_english.append(app)
        
explore_data(Android_english, 0, 3, True)
print('\n')
explore_data(iOS_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

In [13]:
Android_free_apps = []
iOS_free_apps = []
for app in Android_english:
    price = app[7]
    if price == '0' or price == '0.0' or price == '0.00':
        Android_free_apps.append(app)
        
for app in iOS_english:
    price = float(app[4])
    if price == 0.00:
        iOS_free_apps.append(app)
        
print(len(Android_free_apps))
print(len(iOS_free_apps))

8864
3222


Finding an app profile that fits both the App Store and Google Play is crucial for several reasons:

By targeting both iOS and Android platforms, we can reach a wider audience since these two platforms dominate the global smartphone market. This increases the potential user base for your app.

Increased Revenue Potential: With a presence on both platforms, there is more opportunities to monetize apps, whether through in-app purchases, subscriptions, ads, or paid downloads.

Cross-Platform Insights: By analyzing user behavior and preferences on both platforms, we can gather more comprehensive data to inform future updates and improvements to your app.

Brand Consistency: Offering your app on both platforms allows you to maintain brand consistency and recognition, which is important for building trust with users and establishing a strong brand presence.

Risk Mitigation: If there are issues with one platform, such as changes in policies or technical problems, having your app on another platform ensures that your app remains accessible to users.

To validate an app idea effectively, you can follow a strategic approach:



In [14]:
def freq_table(dataset, index):
    f_table = {}
    total = 0
    for app in dataset:
        total += 1
        value = app[index]
        if value in f_table:
            f_table[value] += 1
        else:
            f_table[value] = 1
    percent_f_table = {}
    for key in f_table:
        percent = (f_table[key] / total) * 100
        percent_f_table[key] = percent
    return percent_f_table


In [15]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

In [16]:
display_table(iOS_free_apps, 11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


In the dataset of free English apps, a significant majority, 58.16%, are gaming apps. Entertainment apps make up nearly 8%, with photo and video applications close behind at approximately 5%. Educational apps constitute a smaller fraction, at 3.66%, just slightly more than social networking apps, which represent 3.29% of the dataset.

The data suggests that the App Store, particularly the section for free English apps, tends to favor apps designed for entertainment purposes, such as games, entertainment, photo and video, social networking, sports, and music. In contrast, apps serving more utilitarian functions, like education, shopping, utilities, productivity, and lifestyle, appear less frequently. It's important to note, however, that the prevalence of entertainment-related apps does not necessarily indicate a higher user base, as user demand may not directly correlate with the supply of these apps.


In [17]:
display_table(Android_free_apps, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

On Google Play, the app ecosystem appears to be more oriented towards utility, with a substantial portion of applications falling into categories such as family, tools, business, lifestyle, and productivity. While at first glance, the prevalence of 'family' apps, which constitute nearly 19% of the offerings, might suggest a focus on fun, a closer examination reveals that this category is predominantly composed of children's games. Despite this, the overall landscape on Google Play still shows a stronger emphasis on functional applications when compared to the App Store. This trend towards practicality is further supported by the data presented in the frequency table for the 'Genres' column.

In [18]:
display_table(Android_free_apps, 9)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

In [19]:
genres_ios = freq_table(iOS_free_apps, -5)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in iOS_free_apps:
        genre_app = app[-5]
        if genre_app == genre:            
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


Navigation applications tend to receive a significant number of user reviews, with Waze and Google Maps being notable examples that collectively amass nearly half a million reviews, thereby skewing the average upward for the category.

In [20]:
display_table(Android_free_apps, 5) # the Installs columns

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


In [21]:
categories_android = freq_table(Android_free_apps, 1)

for category in categories_android:
    total = 0
    len_category = 0
    for app in Android_free_apps:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

On average, communication apps exhibit the highest number of installations, standing at an impressive 38,456,119. It's noteworthy that this average is significantly influenced by a select group of applications boasting over one billion installations each—such as WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts. Additionally, several other apps contribute to this skewed average, having achieved notable milestones with over 100 and 500 million installations.

In [22]:
for app in Android_free_apps:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

Excluding all communication apps with installations surpassing 100 million would result in a substantial reduction in the average, approximately by a factor of ten. This adjustment accounts for the removal of the most widely adopted communication applications and offers a more nuanced perspective on the average installations within this category.

In [23]:
for app in Android_free_apps:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

The book and reference genre encompasses a diverse range of applications, including those designed for processing and reading ebooks, collections of libraries, dictionaries, as well as tutorials on programming or languages. However, it's evident that the average within this genre is influenced by a limited number of exceptionally popular apps, which have a notable impact on the overall average installation count. This suggests that a few standout applications significantly contribute to the average installations in this category.

In [24]:
for app in Android_free_apps:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


The observation that the book and reference genre is currently dominated by a select number of highly popular apps indicates that there is untapped potential within this market. To explore this potential further, let's generate app ideas focusing on the segment with moderate popularity, ranging between 1,000,000 and 100,000,000 downloads. This approach seeks to identify opportunities for unique and innovative apps that can thrive in a space not yet dominated by a few major players.

In [25]:
for app in Android_free_apps:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

Given the current landscape, the book and reference niche appears to be predominantly occupied by applications tailored for ebook processing, reading, and diverse library collections and dictionaries. Considering this competitive scenario, it might be prudent to explore alternative app concepts, as developing similar applications may lead to substantial competition within this niche.

Observing the success of apps centered around the Quran indicates the profitability of building an app around a popular book. This insight suggests that transforming a well-known contemporary book into an app could be a lucrative venture for both the Google Play and App Store markets.

Nevertheless, it's important to recognize that the market is saturated with libraries. To differentiate our app, we need to incorporate unique features beyond simply offering the raw version of the book. Potential enhancements might involve daily quotes sourced from the book, an audio version for accessibility, interactive quizzes related to the book's content, a dedicated forum for community discussions, and more. By adding these special features, we aim to provide users with a distinctive and enriched reading experience.